src/peergos/shared/user/fs/DirAccess.java

```diff
@@ -227,58 +239,75 @@ private static PaddedCipherText encryptChildren(SymmetricKey baseKey, List<Capab
             ArrayList<Capability> newFiles = new ArrayList<>(children);
             newFiles.addAll(targetCAPs);
 
-            return Transaction.run(ourPointer.getLocation().owner,
+            return Transaction.run(ourOwner,
                     tid -> withChildren(encryptChildren(ourBaseKey, newFiles))
-                            .commit(ourPointer.getLocation(), signer, network, tid),
+                            .commit(ourOwner, ourPointer.getMapKey(), signer, network, tid),
                     network.dhtClient);
         }
     }
 
-    private CompletableFuture<List<RetrievedCapability>> getNextMetablob(SymmetricKey subfoldersKey, NetworkAccess network) {
+    private CompletableFuture<List<RetrievedCapability>> getNextMetablob(PublicKeyHash owner,
+                                                                         PublicKeyHash writer,
+                                                                         SymmetricKey subfoldersKey,
+                                                                         NetworkAccess network) {
         if (!moreFolderContents.isPresent())
             return CompletableFuture.completedFuture(Collections.emptyList());
-        return network.retrieveAllMetadata(Arrays.asList(moreFolderContents.get().toCapability(subfoldersKey)));
+        Capability cap = moreFolderContents.get().toCapability(subfoldersKey);
+        return network.retrieveAllMetadata(Arrays.asList(new Triple<>(owner, cap.writer.orElse(writer), cap)));
     }
 
-    public CompletableFuture<DirAccess> updateChildLink(Capability ourPointer, RetrievedCapability original,
-                                                        RetrievedCapability modified, SigningPrivateKeyAndPublicHash signer,
-                                                        NetworkAccess network, SafeRandom random) {
-        return removeChild(original, ourPointer, signer, network)
-                .thenCompose(res -> {
-                    return res.addChildAndCommit(modified.capability, ourPointer.baseKey, ourPointer, signer, network, random);
-                });
+    public CompletableFuture<DirAccess> updateChildLink(PublicKeyHash ourOwner,
+                                                        Capability ourPointer,
+                                                        RetrievedCapability original,
+                                                        RetrievedCapability modified,
+                                                        SigningPrivateKeyAndPublicHash signer,
+                                                        NetworkAccess network,
+                                                        SafeRandom random) {
+        return removeChild(original, ourOwner, ourPointer, signer, network)
```

Comment:
do we want an exceptionally clause in here?

Reply:
I think the callers of this should handle that.